### PR TITLE
Fix when to show move arm buttons

### PR DESCRIPTION
**Pull request by OleDrange**

### CODE REVIEW

File: frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx [hunk 1]
```diff
@@ -1,23 +1,42 @@
...
     robot: Robot
     armPosition: string
+    buttonActivated?: boolean
 }
 const feedbackTimer = 10000 // Clear feedback after 10 seconds
 
-export function MoveRobotArm({ robot, armPosition }: RobotProps) {
+export function MoveRobotArm({ robot, armPosition, buttonActivated }: RobotProps) {
     const [feedback, setFeedback] = useState('')
+    const [Usable, setUsable] = useState(!!buttonActivated)
```
Comment (Eddasol) [inline frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx:16]: Lower letter in usable

File: frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx [hunk 2]
```diff
@@ -1,23 +1,42 @@
...
+
+    useEffect(() => {
+        if (typeof buttonActivated !== 'undefined') {
+            setUsable(buttonActivated)
+        }
+    }, [buttonActivated])
+
+    const handleNotUsable = () => {
```
Comment (Eddasol) [inline frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx:24]: suggest renaming to "armButtonStyle"

**Comment by Eddasol** [comment]
The user should get some feedback when the button is disabled. E.g. a popover or popup dialog if clicking the disabled button

File: frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx [hunk 3]
```diff
@@ -1,23 +1,42 @@
...
-import { useState } from 'react'
+import { useState, useEffect } from 'react'
 import { BackendAPICaller } from 'api/ApiCaller'
 import { Robot } from 'models/Robot'
 import { TranslateText } from 'components/Contexts/LanguageContext'
 
 interface RobotProps {
     robot: Robot
     armPosition: string
+    buttonActivated?: boolean
```
Comment (Eddasol) [inline frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx:10]: Do we need this to be optional? If we don't parse a value for buttonActivated the component will always be disabled and not usable anyways

File: frontend/src/components/Pages/RobotPage/RobotPage.tsx [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { Typography } from '@equinor/eds-core-react'
+import { Button, Typography } from '@equinor/eds-core-react'
```
Comment (Eddasol) [inline frontend/src/components/Pages/RobotPage/RobotPage.tsx:1]: Button is not used

File: frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx (hunk 3 above)
Comment (Eddasol) [inline frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx:10]: buttonActivated can be interpreted as button has been clicked/activated. Consider different name.

Possibly name isRobotAvailable?

File: frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx [hunk 5]
```diff
@@ -1,23 +1,42 @@
...
 }
 const feedbackTimer = 10000 // Clear feedback after 10 seconds
 
-export function MoveRobotArm({ robot, armPosition }: RobotProps) {
+export function MoveRobotArm({ robot, armPosition, buttonActivated }: RobotProps) {
     const [feedback, setFeedback] = useState('')
+    const [Usable, setUsable] = useState(!!buttonActivated)
+
+    useEffect(() => {
+        if (typeof buttonActivated !== 'undefined') {
```
Comment (Eddasol) [inline frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx:19]:
```suggestion
        if (buttonActivated) {
```

**Comment by OleDrange** [comment]
> The user should get some feedback when the button is disabled. E.g. a popover or popup dialog if clicking the disabled button

I feel we don't need that. The button goes gray and the mouse pointer shows it is not clickable. Personally, I would be annoyed by a popup for that. Maybe we should get a third opinion? Or some other graphical change on the button itself?

**Comment by Eddasol** [comment]
> I feel we don't need that. The button goes gray and the mouse pointer shows it is not clickable. Personally, I would be annoyed by a popup for that. Maybe we should get a third opinion? Or some other graphical change on the button itself?

Can be popup only for click, and not for hover. This way it's only visible for the confused users. 

Also, most will understand that it's disabled but maybe not why and how to fix it

File: frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx [hunk 6]
```diff
@@ -35,7 +55,9 @@ export function MoveRobotArm({ robot, armPosition }: RobotProps) {
     }
     return (
         <>
-            <Button onClick={onClickMoveArm}>{TranslateText('Set robot arm to ') + '"' + armPosition + '"'}</Button>
+            <Button style={moveArmButtonStyle()} disabled={!usable} onClick={onClickMoveArm}>
```
Comment (Eddasol) [inline frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx:58]:
```suggestion
            <Button disabled={!usable} onClick={onClickMoveArm}>
```
No need with custom disabled style. Happens automatically when disabled

**Comment by Eddasol** [comment]
The button doesn't update to disabled if the robot becomes busy while at robotPage

Can test this by starting a localization mission from the robotPage

**Comment by OleDrange** [comment]
> The button doesn't update to disabled if the robot becomes busy while at robotPage
> 
> Can test this by starting a localization mission from the robotPage

Fixed

**Comment by OleDrange** [comment]
> > I feel we don't need that. The button goes gray and the mouse pointer shows it is not clickable. Personally, I would be annoyed by a popup for that. Maybe we should get a third opinion? Or some other graphical change on the button itself?
> 
> Can be popup only for click, and not for hover. This way it's only visible for the confused users.
> 
> Also, most will understand that it's disabled but maybe not why and how to fix it

Added a popover

File: frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx (hunk 6 above)
Comment (Eddasol) [reply frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx]: If we don't use the built in disable I would use this style: 
"
style = {{
    backgroundColor: tokens.colors.interactive.disabled__fill.hex,
    color: tokens.colors.interactive.disabled__text.hex,
    borderColor: tokens.colors.interactive.disabled__border.hex,
    cursor: "not-allowed",
}}"

to make sure that the style is the same as other disabled buttons

Comment (OleDrange) [reply frontend/src/components/Pages/RobotPage/RobotArmMovement.tsx]: done

**Review by andchiind — APPROVED** [review]
LGTM

**Pull request merged**